kajas/src/app/setup-profile/setup-profile.component.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { LocationService } from '../../services/location.service';
+import { SessionStorageService } from 'angular-web-storage';
+import axios from 'axios';
+
+@Component({
+  selector: 'app-setup-profile',
+  templateUrl: './setup-profile.component.html',
+  styleUrls: ['./setup-profile.component.css']
+})
+export class SetupProfileComponent implements OnInit {
+  profileForm: FormGroup;
+  countries: any[] = [];
+  cities: any[] = [];
+  selectedCountryName: string = '';
+
+  constructor(private fb: FormBuilder, private locationService: LocationService, 
+    private sessionStorage: SessionStorageService) {
+
+    this.profileForm = this.fb.group({
+      id: this.sessionStorage.get('id'),
+      profilePic: [''],
+      bio: [''],
+      firstName: ['', Validators.required],
+      lastName: ['', Validators.required],
+      middleName: [''],
+      email: ['', [Validators.required, Validators.email]],
+      country: ['', Validators.required],
+      city: ['', Validators.required],
+      linkedIn: [''],
+      facebook: [''],
+      instagram: [''],
+      website: [''],
+      kajasUrl: ['']
+    });
+  }
+
+  ngOnInit(): void {
+    this.locationService.getCountries().subscribe(data => {
+      this.countries = data;
+      this.countries.sort((a, b) => a.name.localeCompare(b.name)); 
+    });
+  }
+
+  onCountryChange(event: Event): void {
+    const countryCode = (event.target as HTMLSelectElement).value;
+    this.selectedCountryName = this.countries.find(country => country.id === countryCode).name;
+    this.profileForm.patchValue({ country: this.selectedCountryName });
+
+    this.locationService.getCities(countryCode).subscribe(data => {
+      this.cities = data.filter(city => city['country code'] === countryCode);
+      this.cities.sort((a, b) => a.name.localeCompare(b.name)); // Sort cities alphabetically
+    });
+  }
+
+  async onSubmit(): Promise<void> {
+    console.log(this.profileForm.value)
+    const url = "http://localhost:4000/api/setProfile";
+    const formData = new FormData();
+    Object.keys(this.profileForm.controls).forEach(key => {
+      formData.append(key, this.profileForm.get(key).value);
+    });
+
+    try {
+      const response = await axios.post(url, formData);
+      console.log(response.data);
```

Comment:
### Remove console.logs() and unneccessary codes in the production environment.
- Use console.error() for logging error messages.
- Remove codes/variables that are not used.
https://docs.google.com/presentation/d/1iCoz17mBxZtIQ9PifAW0jjKsuv4xDTlx-TPQxdLySow/edit#slide=id.g2637aafa621_0_0